plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx

```diff
@@ -0,0 +1,264 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, {
+  useRef,
+  useLayoutEffect,
+  useCallback,
+  ReactNode,
+  ReactElement,
+  ComponentPropsWithRef,
+  CSSProperties,
+  UIEventHandler,
+} from 'react';
+import { TableInstance, Hooks, ensurePluginOrder } from 'react-table';
+
+type ReactElementWithChildren<
+  T extends keyof JSX.IntrinsicElements,
+  C extends ReactNode = ReactNode
+> = ReactElement<ComponentPropsWithRef<T> & { children: C }, T>;
+
+type th = ReactElementWithChildren<'th'>;
```

Comment:
`type`s should all be `PascalCase`? so, `th => Th`, `trWithTh => TrWithTh`, `colgroup => ColGroup` etc

Reply:
Updated nonetheless as I can see how it helps differentiate types and variables.